File: Count Number of Nice Subarrays.py
class Solution:
    def numberOfSubarrays(self, nums: List[int], k: int) -> int:
        n = len(nums)
        count = 0
        prefix = []
        subfix = []
        for i, num in enumerate(nums):
            if num % 2:
                prefix.append(count)
                count = 1
            else:
                count += 1

        if count:
            prefix.append(count)

        count = 0
        for i in range(n - 1, -1, -1):
            if nums[i] % 2:
                subfix.append(count)
                count = 1
            else:
                count += 1

        if count:
            subfix.append(count)

        # print(prefix)
        # print(subfix)

        i = k
        j = len(subfix) - 1
        summ = 0
        while i < len(prefix):
            summ += prefix[i] * subfix[j]
            # print(i, j, summ)
            i += 1
            j -= 1

        return summ

File: Count Number of Nice Subarrays.go
func numberOfSubarrays(nums []int, k int) int {
    n := len(nums)
    count := 0
    prefix := make([]int, 0)
    subfix := make([]int, 0)
    for _, num := range nums {
        if num % 2 == 1 {
            prefix = append(prefix, count)
            count = 1
        } else {
            count += 1
        }
    }

    if count > 0 {
        prefix = append(prefix, count)
    }

    count = 0
    for i := n-1; i > -1; i-- {
        if nums[i] % 2 == 1 {
            subfix = append(subfix, count)
            count = 1
        } else {
            count += 1
        }
    }

    if count > 0 {
        subfix = append(subfix, count)
    }

    i, j := k, len(subfix) - 1
    summ := 0
    for i < len(prefix) {
        summ += prefix[i] * subfix[j]
        i++
        j--
    }
    return summ
}

